lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runtime/azbatch/util/batch/AzureBatchHelper.java

```diff
@@ -90,19 +111,47 @@ public void submitJob(final String applicationId, final String storageContainerS
         .withName(SharedAccessSignatureCloudBlobClientProvider.AZURE_STORAGE_CONTAINER_SAS_TOKEN_ENV)
         .withValue(storageContainerSAS);
 
+
+    String portMappings = "";
+
+    Iterator<Integer> iterator = this.portProvider.iterator();
+    while (iterator.hasNext()) {
+      Integer port = iterator.next();
+      System.out.println("iter port is " + port);
+      portMappings += String.format("-p %d:%d ", port, port);
+    }
+
+    TaskContainerSettings containerSettings = null;
+    JobPreparationTask jobPreparationTask = null;
+    if (this.containerRegistry != null) {
+      containerSettings = new TaskContainerSettings()
+          .withRegistry(this.containerRegistry)
+          .withImageName(this.containerImageName)
+          .withContainerRunOptions("-dit --env HOST_IP_ADDR_PATH=$AZ_BATCH_JOB_PREP_DIR/hostip.txt " + portMappings);
+      String captureIpAddressCommandLine =
+          "/bin/bash -c \"rm -f $AZ_BATCH_JOB_PREP_DIR/hostip.txt;" +
+              " echo `hostname -i` > $AZ_BATCH_JOB_PREP_DIR/hostip.txt\"";
+      jobPreparationTask = new JobPreparationTask()
+          .withId("CaptureHostIpAddress")
+          .withCommandLine(captureIpAddressCommandLine);
+    }
+
     JobManagerTask jobManagerTask = new JobManagerTask()
         .withRunExclusive(false)
         .withId(applicationId)
         .withResourceFiles(Collections.singletonList(jarResourceFile))
         .withEnvironmentSettings(Collections.singletonList(environmentSetting))
         .withAuthenticationTokenSettings(authenticationTokenSettings)
+        .withKillJobOnCompletion(false)
+        .withContainerSettings(containerSettings)
```

Comment:
>withContainerSettings [](start = 9, length = 21)

It's OK to set this to NULL? Just wanted to double check, otherwise it's fine.

Reply:
Yes, it is. That is the default. So setting it back to null has no impact.

---
In reply to: [196514414](https://github.com/sharathmalladi/reef/pull/5#discussion_r196514414) [](ancestors = 196514414)